spec/presenters/waste_carriers_engine/registration_presenter_spec.rb

```diff
@@ -0,0 +1,177 @@
+          expect(presenter.complex_organisation_title).to eq("Business name (if applicable)")
+        end
+      end
+    end
+
+    describe "#complex_organisation_name" do
+      let(:registration) do
+        reg = create(:registration, :has_required_data)
+        person = build(:key_person, :has_required_data, :main)
+        person.first_name = "Ryan"
+        person.last_name = "Gosling"
+        reg.key_people.push(person)
+        reg
+      end
+
+      context "when the registration business type is 'partnership'" do
+        it "returns a list of the partners names" do
+          registration.business_type = "partnership"
+          presenter = RegistrationPresenter.new(registration, view)
+          expect(presenter.complex_organisation_name).to eq("Kate Franklin<br>Ryan Gosling")
+        end
+      end
+
+      context "when the registration business type is NOT 'partnership'" do
+        it "returns the company name" do
+          presenter = RegistrationPresenter.new(registration, view)
+          expect(presenter.complex_organisation_name).to eq("Acme Waste")
+        end
+      end
+    end
+
+    describe "#tier_and_registration_type" do
+      let(:registration) { create(:registration, :has_required_data) }
+
+      test_values = {
+        carrier_dealer: "An upper tier waste carrier and dealer",
+        broker_dealer: "An upper tier waste broker and dealer",
+        carrier_broker_dealer: "An upper tier waste carrier, broker and dealer"
+      }
+      test_values.each do |carrier_type, expected|
+        context "when the registration is a '#{carrier_type}'" do
+          it "returns '#{expected}'" do
+            registration.registration_type = carrier_type
+            presenter = RegistrationPresenter.new(registration, view)
+            expect(presenter.tier_and_registration_type).to eq(expected)
+          end
+        end
+      end
+    end
+
+    describe "#expires_after_pluralized" do
+      let(:registration) { create(:registration, :has_required_data) }
+
+      context "when the config is set to 1 year" do
+        before do
+          allow(Rails.configuration).to receive(:expires_after).and_return(1)
+        end
+
+        it "returns '1 year'" do
+          presenter = RegistrationPresenter.new(registration, view)
+          expect(presenter.expires_after_pluralized).to eq("1 year")
+        end
+      end
+
+      context "when the config is set to 3 years" do
+        before do
+          allow(Rails.configuration).to receive(:expires_after).and_return(3)
+        end
+
+        it "returns '3 years'" do
+          presenter = RegistrationPresenter.new(registration, view)
+          expect(presenter.expires_after_pluralized).to eq("3 years")
+        end
+      end
+    end
+
+    describe "#list_main_people" do
+      let(:registration) do
+        reg = create(:registration, :has_required_data)
+        person = build(:key_person, :has_required_data, :main)
```

Comment:
I think this could be done entirely with factories instead of building, modifying and then adding a person. For example:

```
first_man = build(:key_person,
                  :has_required_data, 
                  :main, 
                  first_name: "Ryan", 
                  last_name: "Gosling")
second_man = build(:key_person,
                   :has_required_data,
                   :main,
                   first_name: "Corey",
                   second_name: "Stoll)

create(:registration,
       :has_required_data,
       key_people: [first_man, second_man])
```

Reply:
Just noticed we do the same thing to test `complex_organisation_name` - might be an opportunity to reduce some duplication.